tests/test_xopen.py

```diff
@@ -724,3 +724,20 @@ def test_valid_compression_levels(writer, level, tmpdir):
     with writer(test_file, "wb", level) as handle:
         handle.write(b"test")
     assert gzip.decompress(Path(test_file).read_bytes()) == b"test"
+
+
+def test_format_override(tmpdir):
+    test_file = tmpdir.join("test_gzip_compressed")
+    with xopen(test_file, mode="wb", format="gz") as f:
+        f.write(b"test")
+    test_contents = test_file.read("rb")
+    assert test_contents.startswith(b"\x1f\x8b")  # Gzip magic
+    assert gzip.decompress(test_contents) == b"test"
+
+
+def test_format_override_unsupported_format(tmpdir):
+    test_file = tmpdir.join("test_fairy_format_compressed")
+    with pytest.raises(ValueError) as error:
+        xopen(test_file, mode="wb", format="fairy")
+    assert error.match("not supported")
+    assert error.match("fairy")
```

Comment:
It doesn’t hurt, but it’s inconsistent with the other tests, where you use the `.match` method without an `assert`.